subprojects/ce-login/celogin/include/CeLogin.h

```diff
@@ -142,34 +161,112 @@ struct CeLoginRc
     {}
 
-    operator uint16_t() const
+    inline operator uint16_t() const
     {
         // Format manually to allow for consistant behavior on Big vs Little
         // Endian systems
         return (uint8_t)mComponent << 8 | (uint8_t)mReason;
     }
 
+    inline bool isSuccess() const
+    {
+        return Success == (*this);
+    }
+
     uint8_t mComponent;
     uint8_t mReason;
 };
 
-CeLoginRc getServiceAuthorityV1(
+struct AcfUserFields
+{
+    AcfUserFields() { clear(); }
+
+    void clear()
+    {
+        memset(this, 0x00, sizeof(*this));
+
+        mVersion = CeLoginInvalidVersion;
+        mType = AcfType_Invalid;
+    }
+
+    AcfVersion  mVersion;
+    AcfType     mType;
+    uint64_t    mExpirationTime;
+
+    union
+    {
+        struct
+        {
+            ServiceAuthority mAuth;
+        } mServiceFields;
+
+        struct
+        {
+            char mAdminAuthCode[AdminAuthCodeMaxLen];
+            uint64_t mAdminAuthCodeLength;
+        } mAdminResetFields;
+    };
+};
+
+/// @note This function will return failure if called with a V2 ACF
+[[deprecated]] CeLoginRc getServiceAuthorityV1(
     const uint8_t* accessControlFileParm,
     const uint64_t accessControlFileLengthParm, const char* passwordParm,
     const uint64_t passwordLengthParm,
-    const uint64_t timeSinceUnixEpocInSecondsParm, const uint8_t* publicKeyParm,
+    const uint64_t timeSinceUnixEpochInSecondsParm, const uint8_t* publicKeyParm,
     const uint64_t publicKeyLengthParm, const char* serialNumberParm,
     const uint64_t serialNumberLengthParm, ServiceAuthority& authorityParm,
     uint64_t& expirationTimeParm);
 
-CeLoginRc checkServiceAuthorityAcfIntegrityV1(
+/// @note This function will return failure if called with a V2 ACF
+[[deprecated]] CeLoginRc checkServiceAuthorityAcfIntegrityV1(
     const uint8_t* accessControlFileParm,
     const uint64_t accessControlFileLengthParm,
-    const uint64_t timeSinceUnixEpocInSecondsParm, const uint8_t* publicKeyParm,
+    const uint64_t timeSinceUnixEpochInSecondsParm, const uint8_t* publicKeyParm,
     const uint64_t publicKeyLengthParm, const char* serialNumberParm,
     const uint64_t serialNumberLengthParm, ServiceAuthority& authorityParm,
     uint64_t& expirationTimeParm);
 
+/** @brief Validate an ACF file and parse out the relevant fields
+ *
+ *  This function decodes and verifies several aspects of the provided ACF file.
+ *  This includes verifying the signature over the ACF, ensuring the ACF is not expired,
+ *  checking the serial number fields, verifying the correct password was entered (for
+ *  a service ACF) and validating the replay ID, if it exists. Upon successful validation,
+ *  a AcfUserFields object is returned with the fields relevant to the BMC.
+ *
+ *  @param accessControlFileParm a pointer to the ASN1 encoded binary ACF
+ *  @param accessControlFileLengthParm the length of the provided ACF
+ *  @param passwordParm a pointer to the provided password
```

Comment:
Ben suggested that interface was not needed; I can easily add that public interface back into the code.

Reply:
For admin-reset targeted ACF support I call CeLogin::checkAuthorizationAndGetAcfUserFieldsV2(). If this is successful I check for type admin-reset. If type is admin-reset (populated by call) I reset the admin password using the admin hash value (populated by the call) and update the replayId (populated by the call).

Reply:
Ah, this works because there is no password required to validate the admin-reset ACF. But V2 service ACFs will need the early validation without a specified password. So I think the interface needs to be added back.

Reply:
But I don't understand how you know what type of ACF it is at the time of upload.  You shouldn't be reaching into the ACF without doing proper verification or you open us up to possible malicious injection

Reply:
As the code sits now checkAcfIntegrityV2() is a subset of checkAuthorizationAndGetAcfUserFieldsV2(). 
Both of these will pass/fail on signature, expiration date and serial number.
Additionally there are two things we need:
1. A way to get ACF type without password.
2. A way to verify user with password but without replay Id.
What I have proposed and tested (limited testing) to cover the above two scenarios is:
1. checkAuthorizationAndGetAcfUserFieldsV2() changed to populate user fields (includes ACF type) and return GetSevAuth_InvalidPasswordPtr or GetSevAuth_InvalidPasswordLength when no password is given.
2. I can pass zero for replayID when authenticating with password. Zero should always pass replay check.
My logic is:
1. When I am installing an ACF (includes admin-reset feature) I will pass a valid replayID and no password.
2. When I am authenticating a user I will give a password and zero replayID.
This makes checkAcfIntegrityV2() unnecessary since the same  checks are done in AuthorizationAndGetAcfUserFieldsV2() and user fields are returned if this function completes.

Reply:
List item got "smart". 1 and 2 are the scenarios we need to handle. 3 and 4 are the ways to handle 1 and 2. 5 and 6 are the logic behind 2 and 3.

Reply:
I will take a stab at redefining these interfaces.

Reply:
checkAcfIntegrityV2 does make sense for getting the expiration date which is needed for a higher level call (getAcfInfo). So I can still use it.
One other thing I thought of if we want to prevent existting ACF V2 from being overwritten by the install of an ACF V1 then we need to be able to determine version without password and zero-replay-ID. Falls into the same category of need some (most) user info without password/replay.

Reply:
The latest changes to the interfaces and the additional interfaces address the concerns that I raised.